include/ESA.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef ESA_HPP
+#define ESA_HPP
+
+#include <limits>
```

Comment:
Make sure that these #include are really necessary. To check if they are necessary remove them and try to recompile